License plate model

12gabriel3 opened the pull request:
Add license plate model. Same as https://github.com/EveronLife/EveronLife/pull/108 but with a more detailed model and using MatPBRMulti. Dirt and mud can be easily changed with a slider.

![image](https://user-images.githubusercontent.com/16885097/173265006-3651f4f6-8e2a-4065-be4b-9a9e80709df6.png)

Arkensor reviewed (changes requested):
I removed the modified vehicle prefabs from testing that were accidentally committed as well.

There are two layout files right now under `UI/Layouts/LicensePlate`. Which one of them is used? I think I saw the non-prefixed version referenced now, but the majority of changes you did were to that? Do we need both? If not please select the one you intended to use inside the `LicensePlate.et` and delete the other please.

ThibaudetTevai commented on src/Scripts/Game/Features/LicensePlate/EL_LicensePlateGeneratorGeneric.c:
```diff
@@ -5,9 +5,8 @@ class EL_LicensePlateGeneratorGeneric: EL_LicensePlateGeneratorBase
 	{
 		string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
```
```suggestion
		const string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
```

ThibaudetTevai commented on src/Scripts/Game/Features/LicensePlate/EL_LicensePlateGeneratorGeneric.c:
```diff
@@ -5,9 +5,8 @@ class EL_LicensePlateGeneratorGeneric: EL_LicensePlateGeneratorBase
 	{
 		string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
 		string p1 = alphabet.Get(Math.RandomInt(0, 25)) + alphabet.Get(Math.RandomInt(0, 25));
```
```suggestion
		string p1 = alphabet.Get(Math.RandomInt(0, alphabet.Length() - 1)) + alphabet.Get(Math.RandomInt(0, alphabet.Length() - 1));
```

12gabriel3 replied:
Done in https://github.com/EveronLife/EveronLife/pull/121/commits/c3a1d7bc3e043d691cd0b3d90c1723d1fc9c63cb

ThibaudetTevai reviewed (changes requested):
Review done .
Just add some suggestion on [EL_LicensePlateGeneratorGeneric.c](https://github.com/EveronLife/EveronLife/pull/121/files#diff-7146aa0f07c7511521d7812c457275ea6a53275b448a2bb85721fd25856107a0)

12gabriel3 commented:
> I removed the modified vehicle prefabs from testing that were accidentally committed.
> 
> There are two layout files right now under `UI/Layouts/LicensePlate`. Which one of them is used? I think I saw the non-prefixed version referenced now, but the majority of changes you did were to that? Do we need both? If not please select the one you intended to use inside the `LicensePlate.et` and delete the other please.

The original license plate used 2 layouts as well,

> I removed the modified vehicle prefabs from testing that were accidentally committed.
> 
> There are two layout files right now under `UI/Layouts/LicensePlate`. Which one of them is used? I think I saw the non-prefixed version referenced now, but the majority of changes you did were to that? Do we need both? If not please select the one you intended to use inside the `LicensePlate.et` and delete the other please.

I think we need both. They were added both to Everon Life main branch in https://github.com/EveronLife/EveronLife/commit/91bf5ac279f5b1c3540b30f5f9316e9752c78812

Maybe @Thurston00 or @Jacob-Mango could confirm this.

Arkensor commented:
Something isn't quite right with this @12gabriel3 I keep getting this error
```
    ENTITY       : Create entity @"ENTITY:2305843009213693972" ('EL_LicensePlateEntity') at <0.000000 0.784300 1.681000>
     RESOURCES (E): Can't remap 'Assets/Vehicles/LicensePlate/LicensePlate.xob' material from 'PIPMaterial' to 'Assets/Vehicles/LicensePlate/data/LicensePlateMaterial.emat'. Source material do not exist
```

I am not sure why it keeps looking for a PIPMaterial. I could not find it. Also right now only the `LicensePlate.layout` is used. So I was wondering why you even did any edits to the other one? I will fix that and some other little thing I wanted to do when already at this, but we need to get the license plates to work without the error first.
